tests/integration_tests/conftest.py

```diff
@@ -131,28 +143,35 @@ def setup_image(session_cloud: IntegrationCloud, request):
     request.addfinalizer(session_cloud.delete_snapshot)
 
 
-def _collect_logs(
-    instance: IntegrationInstance, node_id: str, test_failed: bool
-):
-    """Collect logs from remote instance.
-
-    Args:
-        instance: The current IntegrationInstance to collect logs from
-        node_id: The pytest representation of this test, E.g.:
-            tests/integration_tests/test_example.py::TestExample.test_example
-        test_failed: If test failed or not
-    """
-    if any(
-        [
-            integration_settings.COLLECT_LOGS == "NEVER",
-            integration_settings.COLLECT_LOGS == "ON_ERROR"
-            and not test_failed,
-        ]
-    ):
-        return
+def _collect_logs(instance: IntegrationInstance, log_dir: Path):
```

Comment:
To help read the following sections:
Instead of `_collect_logs`, we instead call a new `_collect_artifacts` after every test. This function will check what needs to be setup based on the integration settings, then call `_collect_logs` and/or `_collect_coverage` accordingly.